l1-contracts/test/Outbox.t.sol

```diff
@@ -1,135 +1,254 @@
 // SPDX-License-Identifier: Apache-2.0
-// Copyright 2023 Aztec Labs.
+// Copyright 2024 Aztec Labs.
 pragma solidity >=0.8.18;
 
 import {Test} from "forge-std/Test.sol";
-import {IOutbox} from "../src/core/interfaces/messagebridge/IOutbox.sol";
 import {Outbox} from "../src/core/messagebridge/Outbox.sol";
-import {Registry} from "../src/core/messagebridge/Registry.sol";
+import {IOutbox} from "../src/core/interfaces/messagebridge/IOutbox.sol";
 import {Errors} from "../src/core/libraries/Errors.sol";
 import {DataStructures} from "../src/core/libraries/DataStructures.sol";
-import {MessageBox} from "../src/core/libraries/MessageBox.sol";
+import {Hash} from "../src/core/libraries/Hash.sol";
+import {NaiveMerkle} from "./merkle/Naive.sol";
+import {MerkleTestUtil} from "./merkle/TestUtil.sol";
 
 contract OutboxTest is Test {
-  Registry internal registry;
-  Outbox internal outbox;
-  uint256 internal version = 0;
+  using Hash for DataStructures.L2ToL1Msg;
 
-  event MessageAdded(bytes32 indexed entryKey);
-  event MessageConsumed(bytes32 indexed entryKey, address indexed recipient);
+  address internal constant ROLLUP_CONTRACT = address(0x42069123);
+  address internal constant NOT_ROLLUP_CONTRACT = address(0x69);
+  address internal constant NOT_RECIPIENT = address(0x420);
+  uint256 internal constant DEFAULT_TREE_HEIGHT = 2;
+  uint256 internal constant AZTEC_VERSION = 1;
+
+  Outbox internal outbox;
+  NaiveMerkle internal zeroedTree;
+  MerkleTestUtil internal merkleTestUtil;
 
   function setUp() public {
-    address rollup = address(this);
-    registry = new Registry();
-    outbox = new Outbox(address(registry));
-    version = registry.upgrade(rollup, address(0x0), address(outbox));
+    outbox = new Outbox(ROLLUP_CONTRACT);
+    zeroedTree = new NaiveMerkle(DEFAULT_TREE_HEIGHT);
+    merkleTestUtil = new MerkleTestUtil();
   }
 
-  function _fakeMessage() internal view returns (DataStructures.L2ToL1Msg memory) {
+  function _fakeMessage(address _recipient) internal view returns (DataStructures.L2ToL1Msg memory) {
     return DataStructures.L2ToL1Msg({
       sender: DataStructures.L2Actor({
         actor: 0x2000000000000000000000000000000000000000000000000000000000000000,
-        version: version
+        version: AZTEC_VERSION
       }),
-      recipient: DataStructures.L1Actor({actor: address(this), chainId: block.chainid}),
+      recipient: DataStructures.L1Actor({actor: _recipient, chainId: block.chainid}),
       content: 0x3000000000000000000000000000000000000000000000000000000000000000
     });
   }
 
   function testRevertIfInsertingFromNonRollup() public {
```

Comment:
To use the fuzzer here, you can add an `address _caller` in the parameters, and then `vm.assume(ROLLUP_CONTRACT != _caller);`. Since there are only one case where this would make us fail, it is not an issue with looping until hitting something useful.